src/sif/truckcost.cc

```diff
@@ -381,7 +381,7 @@ bool TruckCost::AllowMultiPass() const {
 bool TruckCost::ModeSpecificAllowed(const baldr::AccessRestriction& restriction) const {
   switch (restriction.type()) {
     case AccessType::kHazmat:
-      if (hazmat_ != restriction.value()) {
+      if (hazmat_ && !restriction.value()) {
```

Comment:
first check if hazmat is even turned on in the request before comparing it, since otherwise would not allow a combo like hazmat: false in the request but hazmat: true on the road.